File: backend/app/api/v1/investor_metrics.py
"""
Investor Metrics API Endpoints
Dashboard metrics for CAC, LTV, GMV, NRR, Churn
ADMIN ONLY
"""
from fastapi import APIRouter, Depends, Query
from typing import Optional, Dict, Any
from datetime import datetime, timezone, timedelta

from app.core.security import get_current_admin
from app.db.connection import get_database
from app.models.database import User
from app.services.investor_metrics_service import InvestorMetricsService
from pydantic import BaseModel


router = APIRouter(prefix="/metrics", tags=["Investor Metrics"])


class MetricsRequest(BaseModel):
    """Request for metrics calculation"""
    period_start: datetime
    period_end: datetime
    marketing_spend: Optional[Dict[str, float]] = None


@router.post("/cac", dependencies=[Depends(get_current_admin)])
async def calculate_cac_metric(
    request: MetricsRequest,
    db=Depends(get_database),
):
    """
    Calculate Customer Acquisition Cost
    
    ADMIN ONLY
    
    Formula: Total Marketing Spend / New Customers
    
    Request body:
    ```json
    {
        "period_start": "2024-01-01T00:00:00Z",
        "period_end": "2024-01-31T23:59:59Z",
        "marketing_spend": {
            "google_ads": 50000,
            "facebook": 30000,
            "referral": 10000,
            "organic": 0
        }
    }
    ```
    """
    marketing_spend = request.marketing_spend or {
        "google_ads": 0,
        "facebook": 0,
        "referral": 0,
        "organic": 0
    }
    
    cac = await InvestorMetricsService.calculate_cac(
        db,
        request.period_start,
        request.period_end,
        marketing_spend
    )
    
    return {
        "cac": cac.model_dump(by_alias=True),
        "summary": {
            "blended_cac": round(cac.blended_cac, 2),
            "total_users": cac.total_new_users,
            "total_spend": cac.total_marketing_spend,
            "efficiency": f"{cac.cac_efficiency:.1%}",
            "status": "good" if cac.cac_efficiency >= 1.0 else "needs_improvement"
        }
    }


@router.get("/ltv/{cohort_month}", dependencies=[Depends(get_current_admin)])
async def calculate_ltv_metric(
    cohort_month: str,
    db=Depends(get_database),
):
    """
    Calculate Lifetime Value for a cohort
    
    ADMIN ONLY
    
    Args:
        cohort_month: Format "YYYY-MM" (e.g., "2024-01")
    
    Formula: Avg Revenue × Avg Lifespan × Gross Margin
    """
    ltv = await InvestorMetricsService.calculate_ltv(db, cohort_month)
    
    ltv_cac_status = "excellent" if ltv.ltv_cac_ratio and ltv.ltv_cac_ratio >= 3.0 else "needs_improvement"
    
    return {
        "ltv": ltv.model_dump(by_alias=True),
        "summary": {
            "ltv": round(ltv.ltv, 2),
            "cac": round(ltv.cac, 2) if ltv.cac else None,
            "ltv_cac_ratio": round(ltv.ltv_cac_ratio, 2) if ltv.ltv_cac_ratio else None,
            "cohort_size": ltv.cohort_size,
            "retention_month_6": f"{ltv.retention_rate_month_6:.1%}",
            "status": ltv_cac_status
        }
    }


@router.post("/gmv", dependencies=[Depends(get_current_admin)])
async def calculate_gmv_metric(
    request: MetricsRequest,
    db=Depends(get_database),
):
    """
    Calculate Gross Merchandise Value
    
    ADMIN ONLY
    
    GMV = Total booking value processed
    Net Revenue = GMV × Take Rate (20%)
    """
    gmv = await InvestorMetricsService.calculate_gmv(
        db,
        request.period_start,
        request.period_end
    )
    
    return {
        "gmv": gmv.model_dump(by_alias=True),
        "summary": {
            "total_gmv": round(gmv.total_gmv, 2),
            "completed_gmv": round(gmv.completed_gmv, 2),
            "platform_revenue": round(gmv.platform_fee_collected, 2),
            "take_rate": f"{gmv.take_rate_percentage:.1f}%",
            "growth_rate": f"{gmv.gmv_growth_rate:+.1f}%",
            "avg_booking": round(gmv.avg_booking_value, 2)
        }
    }


@router.post("/repeat-rate", dependencies=[Depends(get_current_admin)])
async def calculate_repeat_booking_metric(
    request: MetricsRequest,
    db=Depends(get_database),
):
    """
    Calculate Repeat Booking Rate
    
    ADMIN ONLY
    
    Key indicator of product-market fit
    
    Target: >40% repeat rate
    """
    repeat = await InvestorMetricsService.calculate_repeat_booking_metrics(
        db,
        request.period_start,
        request.period_end
    )
    
    pmf_status = "strong" if repeat.repeat_rate_percentage >= 40 else "developing"
    
    return {
        "repeat_metrics": repeat.model_dump(by_alias=True),
        "summary": {
            "repeat_rate": f"{repeat.repeat_rate_percentage:.1f}%",
            "total_users": repeat.total_grihastas,
            "repeat_users": repeat.repeat_bookers,
            "repeat_revenue_pct": f"{repeat.repeat_revenue_percentage:.1f}%",
            "pmf_indicator": pmf_status
        }
    }


@router.post("/churn", dependencies=[Depends(get_current_admin)])
async def calculate_churn_metric(
    request: MetricsRequest,
    db=Depends(get_database),
):
    """
    Calculate Acharya Churn Rate
    
    ADMIN ONLY
    
    Churn = Acharyas with 0 bookings for 60+ days
    
    Target: <5% monthly churn
    """
    churn = await InvestorMetricsService.calculate_acharya_churn(
        db,
        request.period_start,
        request.period_end
    )
    
    health_status = "healthy" if churn.monthly_churn_rate < 5.0 else "at_risk"
    
    return {
        "churn_metrics": churn.model_dump(by_alias=True),
        "summary": {
            "total_acharyas": churn.total_acharyas,
            "active_acharyas": churn.active_acharyas,
            "churn_rate": f"{churn.monthly_churn_rate:.1f}%",
            "churned_count": churn.churned_acharyas,
            "supply_health": health_status
        }
    }


@router.get("/dashboard", dependencies=[Depends(get_current_admin)])
async def get_investor_dashboard(
    days: int = Query(30, description="Period in days"),
    db=Depends(get_database),
):
    """
    Get comprehensive investor dashboard metrics
    
    ADMIN ONLY
    
    Returns:
        All key metrics for the specified period
    """
    period_end = datetime.now(timezone.utc)
    period_start = period_end - timedelta(days=days)
    
    # Calculate all metrics in parallel (simplified - would use asyncio.gather)
    gmv = await InvestorMetricsService.calculate_gmv(db, period_start, period_end)
    repeat = await InvestorMetricsService.calculate_repeat_booking_metrics(db, period_start, period_end)
    churn = await InvestorMetricsService.calculate_acharya_churn(db, period_start, period_end)
    
    # Get recent LTV (last month's cohort)
    last_month = period_end - timedelta(days=30)
    cohort_month = f"{last_month.year}-{last_month.month:02d}"
    try:
        ltv = await InvestorMetricsService.calculate_ltv(db, cohort_month)
        ltv_data = ltv.model_dump(by_alias=True)
    except:
        ltv_data = None
    
    return {
        "period": {
            "start": period_start,
            "end": period_end,
            "days": days
        },
        "gmv": {
            "total": round(gmv.total_gmv, 2),
            "completed": round(gmv.completed_gmv, 2),
            "growth_rate": round(gmv.gmv_growth_rate, 2),
            "platform_revenue": round(gmv.platform_fee_collected, 2),
            "avg_booking_value": round(gmv.avg_booking_value, 2)
        },
        "repeat_bookings": {
            "repeat_rate": round(repeat.repeat_rate_percentage, 2),
            "total_users": repeat.total_grihastas,
            "repeat_users": repeat.repeat_bookers,
            "repeat_revenue_pct": round(repeat.repeat_revenue_percentage, 2)
        },
        "supply_health": {
            "total_acharyas": churn.total_acharyas,
            "active_acharyas": churn.active_acharyas,
            "churn_rate": round(churn.monthly_churn_rate, 2),
            "inactive_count": churn.inactive_acharyas
        },
        "ltv_analysis": ltv_data,
        "health_indicators": {
            "gmv_growth": "growing" if gmv.gmv_growth_rate > 0 else "declining",
            "pmf_strength": "strong" if repeat.repeat_rate_percentage >= 40 else "developing",
            "supply_stability": "stable" if churn.monthly_churn_rate < 5 else "at_risk",
            "ltv_cac_ratio": round(ltv.ltv_cac_ratio, 2) if ltv_data and ltv.ltv_cac_ratio else None
        }
    }


@router.get("/trends", dependencies=[Depends(get_current_admin)])
async def get_metric_trends(
    metric: str = Query(..., description="Metric name: gmv, repeat_rate, churn"),
    months: int = Query(6, description="Number of months to show"),
    db=Depends(get_database),
):
    """
    Get historical trends for a metric
    
    ADMIN ONLY
    
    Useful for charts/graphs in admin dashboard
    """
    trends = []
    current_date = datetime.now(timezone.utc)
    
    for i in range(months, 0, -1):
        month_start = current_date.replace(day=1) - timedelta(days=30 * i)
        month_end = month_start + timedelta(days=30)
        
        if metric == "gmv":
            data = await InvestorMetricsService.calculate_gmv(db, month_start, month_end)
            value = data.total_gmv
        elif metric == "repeat_rate":
            data = await InvestorMetricsService.calculate_repeat_booking_metrics(db, month_start, month_end)
            value = data.repeat_rate_percentage
        elif metric == "churn":
            data = await InvestorMetricsService.calculate_acharya_churn(db, month_start, month_end)
            value = data.monthly_churn_rate
        else:
            value = 0
        
        trends.append({
            "month": f"{month_start.year}-{month_start.month:02d}",
            "value": round(value, 2)
        })
    
    return {
        "metric": metric,
        "trends": trends
    }

File: admin-savitara-web/pages/investor-metrics.js
import { useState, useEffect } from 'react';
import Head from 'next/head';
import {
  Container,
  Typography,
  Paper,
  Box,
  Card,
  CardContent,
  Grid,
  CircularProgress,
  FormControl,
  InputLabel,
  Select,
  MenuItem,
  Stack,
  Alert,
  Snackbar,
  Button,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
} from '@mui/material';
import {
  TrendingUp,
  MonetizationOn,
  People,
  Repeat,
  ShowChart,
  Refresh,
  Download,
} from '@mui/icons-material';
import { format, subDays } from 'date-fns';
import {
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend,
  ResponsiveContainer,
} from 'recharts';
import Layout from '../src/components/Layout';
import withAuth from '../src/hoc/withAuth';
import { adminAPI } from '../src/services/api';

function InvestorMetrics() {
  const [loading, setLoading] = useState(true);
  const [periodDays, setPeriodDays] = useState(30);
  const [dashboard, setDashboard] = useState(null);
  const [gmvTrends, setGmvTrends] = useState([]);
  const [cacData, setCacData] = useState(null);
  const [ltvData, setLtvData] = useState(null);
  const [snackbar, setSnackbar] = useState({ open: false, message: '', severity: 'success' });

  useEffect(() => {
    fetchDashboard();
    fetchTrends();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [periodDays]);

  const fetchDashboard = async () => {
    setLoading(true);
    try {
      const response = await adminAPI.get(`/metrics/dashboard?days=${periodDays}`);
      setDashboard(response.data);
      
      // Fetch CAC for the period
      const period_end = new Date().toISOString();
      const period_start = subDays(new Date(), periodDays).toISOString();
      
      const cacResponse = await adminAPI.post('/metrics/cac', {
        period_start,
        period_end,
        marketing_spend: {
          google_ads: 50000,
          facebook: 30000,
          referral: 10000,
          organic: 0,
        },
      });
      setCacData(cacResponse.data);
      
      // Fetch LTV for current month's cohort
      const cohort_month = format(new Date(), 'yyyy-MM');
      const ltvResponse = await adminAPI.get(`/metrics/ltv/${cohort_month}`);
      setLtvData(ltvResponse.data);
      
    } catch (error) {
      console.error('Failed to load investor metrics:', error);
      showSnackbar(error.response?.data?.message || 'Failed to load investor metrics', 'error');
    } finally {
      setLoading(false);
    }
  };

  const fetchTrends = async () => {
    try {
      const response = await adminAPI.get('/metrics/trends?metric=gmv&months=6');
      setGmvTrends(response.data.trends);
    } catch (error) {
      console.error('Failed to load trends:', error);
    }
  };

  const formatCurrency = (value) => {
    return new Intl.NumberFormat('en-IN', {
      style: 'currency',
      currency: 'INR',
      maximumFractionDigits: 0,
    }).format(value);
  };

  const formatPercentage = (value) => {
    return `${value.toFixed(1)}%`;
  };

  const getHealthColor = (indicator) => {
    const colorMap = {
      growing: 'success.main',
      strong: 'success.main',
      stable: 'success.main',
      excellent: 'success.main',
      declining: 'error.main',
      developing: 'warning.main',
      at_risk: 'error.main',
    };
    return colorMap[indicator] || 'text.secondary';
  };

  const showSnackbar = (message, severity = 'success') =>
    setSnackbar({ open: true, message, severity });

  const closeSnackbar = () => setSnackbar({ ...snackbar, open: false });

  const handleExportReport = () => {
    // Generate CSV report
    const csvData = [
      ['Savitara Investor Metrics Report'],
      ['Period', `Last ${periodDays} days`],
      ['Generated', format(new Date(), 'yyyy-MM-dd HH:mm')],
      [],
      ['Key Metrics'],
      ['GMV', dashboard?.gmv?.total || 0],
      ['Platform Revenue', dashboard?.gmv?.platform_revenue || 0],
      ['GMV Growth Rate', `${dashboard?.gmv?.growth_rate || 0}%`],
      [],
      ['Repeat Bookings'],
      ['Repeat Rate', `${dashboard?.repeat_bookings?.repeat_rate || 0}%`],
      ['Total Users', dashboard?.repeat_bookings?.total_users || 0],
      ['Repeat Users', dashboard?.repeat_bookings?.repeat_users || 0],
      [],
      ['Supply Health'],
      ['Total Acharyas', dashboard?.supply_health?.total_acharyas || 0],
      ['Active Acharyas', dashboard?.supply_health?.active_acharyas || 0],
      ['Churn Rate', `${dashboard?.supply_health?.churn_rate || 0}%`],
    ];
    
    const csvContent = csvData.map(row => row.join(',')).join('\n');
    const blob = new Blob([csvContent], { type: 'text/csv' });
    const url = URL.createObjectURL(blob);
    const a = document.createElement('a');
    a.href = url;
    a.download = `savitara-investor-metrics-${format(new Date(), 'yyyy-MM-dd')}.csv`;
    a.click();
    URL.revokeObjectURL(url);
  };

  return (
    <Layout>
      <Head>
        <title>Investor Metrics - Savitara Admin</title>
      </Head>

      <Container maxWidth="xl" sx={{ mt: 4, mb: 4 }}>
        <Stack direction="row" justifyContent="space-between" alignItems="center" sx={{ mb: 3 }}>
          <Typography variant="h4" sx={{ display: 'flex', alignItems: 'center', gap: 1 }}>
            <ShowChart /> Investor Metrics Dashboard
          </Typography>
          <Stack direction="row" spacing={2}>
            <FormControl size="small" sx={{ minWidth: 150 }}>
              <InputLabel>Period</InputLabel>
              <Select
                value={periodDays}
                label="Period"
                onChange={(e) => setPeriodDays(e.target.value)}
              >
                <MenuItem value={7}>Last 7 days</MenuItem>
                <MenuItem value={30}>Last 30 days</MenuItem>
                <MenuItem value={90}>Last 90 days</MenuItem>
                <MenuItem value={180}>Last 6 months</MenuItem>
              </Select>
            </FormControl>
            <Button
              variant="outlined"
              startIcon={<Refresh />}
              onClick={fetchDashboard}
            >
              Refresh
            </Button>
            <Button
              variant="contained"
              startIcon={<Download />}
              onClick={handleExportReport}
              disabled={!dashboard}
            >
              Export Report
            </Button>
          </Stack>
        </Stack>

        {loading ? (
          <Box sx={{ display: 'flex', justifyContent: 'center', mt: 10 }}>
            <CircularProgress />
          </Box>
        ) : (
          <>
            {/* GMV Metrics */}
            <Typography variant="h6" gutterBottom sx={{ mt: 3 }}>
              Gross Merchandise Value (GMV)
            </Typography>
            <Grid container spacing={3}>
              <Grid item xs={12} sm={6} md={3}>
                <Card>
                  <CardContent>
                    <Stack direction="row" justifyContent="space-between" alignItems="center">
                      <Box>
                        <Typography color="text.secondary" variant="body2">
                          Total GMV
                        </Typography>
                        <Typography variant="h5">
                          {formatCurrency(dashboard?.gmv?.total || 0)}
                        </Typography>
                      </Box>
                      <MonetizationOn color="primary" sx={{ fontSize: 40 }} />
                    </Stack>
                  </CardContent>
                </Card>
              </Grid>
              <Grid item xs={12} sm={6} md={3}>
                <Card>
                  <CardContent>
                    <Stack direction="row" justifyContent="space-between" alignItems="center">
                      <Box>
                        <Typography color="text.secondary" variant="body2">
                          Platform Revenue
                        </Typography>
                        <Typography variant="h5">
                          {formatCurrency(dashboard?.gmv?.platform_revenue || 0)}
                        </Typography>
                      </Box>
                      <MonetizationOn color="success" sx={{ fontSize: 40 }} />
                    </Stack>
                  </CardContent>
                </Card>
              </Grid>
              <Grid item xs={12} sm={6} md={3}>
                <Card>
                  <CardContent>
                    <Stack direction="row" justifyContent="space-between" alignItems="center">
                      <Box>
                        <Typography color="text.secondary" variant="body2">
                          Growth Rate
                        </Typography>
                        <Typography
                          variant="h5"
                          sx={{ color: getHealthColor(dashboard?.health_indicators?.gmv_growth) }}
                        >
                          {formatPercentage(dashboard?.gmv?.growth_rate || 0)}
                        </Typography>
                      </Box>
                      <TrendingUp
                        sx={{
                          fontSize: 40,
                          color: getHealthColor(dashboard?.health_indicators?.gmv_growth),
                        }}
                      />
                    </Stack>
                  </CardContent>
                </Card>
              </Grid>
              <Grid item xs={12} sm={6} md={3}>
                <Card>
                  <CardContent>
                    <Stack direction="row" justifyContent="space-between" alignItems="center">
                      <Box>
                        <Typography color="text.secondary" variant="body2">
                          Avg Booking Value
                        </Typography>
                        <Typography variant="h5">
                          {formatCurrency(dashboard?.gmv?.avg_booking_value || 0)}
                        </Typography>
                      </Box>
                      <MonetizationOn color="info" sx={{ fontSize: 40 }} />
                    </Stack>
                  </CardContent>
                </Card>
              </Grid>
            </Grid>

            {/* LTV & CAC Metrics */}
            <Typography variant="h6" gutterBottom sx={{ mt: 4 }}>
              Customer Metrics
            </Typography>
            <Grid container spacing={3}>
              <Grid item xs={12} sm={6} md={4}>
                <Card>
                  <CardContent>
                    <Typography color="text.secondary" gutterBottom>
                      Lifetime Value (LTV)
                    </Typography>
                    <Typography variant="h4">
                      {ltvData ? formatCurrency(ltvData.summary.ltv) : '-'}
                    </Typography>
                    <Typography variant="body2" color="text.secondary" sx={{ mt: 1 }}>
                      Current month cohort
                    </Typography>
                  </CardContent>
                </Card>
              </Grid>
              <Grid item xs={12} sm={6} md={4}>
                <Card>
                  <CardContent>
                    <Typography color="text.secondary" gutterBottom>
                      Customer Acquisition Cost (CAC)
                    </Typography>
                    <Typography variant="h4">
                      {cacData ? formatCurrency(cacData.summary.blended_cac) : '-'}
                    </Typography>
                    <Typography variant="body2" color="text.secondary" sx={{ mt: 1 }}>
                      Blended across all channels
                    </Typography>
                  </CardContent>
                </Card>
              </Grid>
              <Grid item xs={12} sm={6} md={4}>
                <Card>
                  <CardContent>
                    <Typography color="text.secondary" gutterBottom>
                      LTV:CAC Ratio
                    </Typography>
                    <Typography
                      variant="h4"
                      sx={{ color: getHealthColor(ltvData?.summary?.status) }}
                    >
                      {ltvData?.summary?.ltv_cac_ratio?.toFixed(2) || '-'}
                    </Typography>
                    <Typography variant="body2" color="text.secondary" sx={{ mt: 1 }}>
                      Target: ≥ 3.0 (Excellent)
                    </Typography>
                  </CardContent>
                </Card>
              </Grid>
            </Grid>

            {/* Repeat Bookings & Churn */}
            <Typography variant="h6" gutterBottom sx={{ mt: 4 }}>
              Engagement & Retention
            </Typography>
            <Grid container spacing={3}>
              <Grid item xs={12} sm={6} md={3}>
                <Card>
                  <CardContent>
                    <Stack direction="row" justifyContent="space-between" alignItems="center">
                      <Box>
                        <Typography color="text.secondary" variant="body2">
                          Repeat Booking Rate
                        </Typography>
                        <Typography
                          variant="h5"
                          sx={{
                            color: getHealthColor(dashboard?.health_indicators?.pmf_strength),
                          }}
                        >
                          {formatPercentage(dashboard?.repeat_bookings?.repeat_rate || 0)}
                        </Typography>
                        <Typography variant="caption" color="text.secondary">
                          Target: ≥ 40% (Strong PMF)
                        </Typography>
                      </Box>
                      <Repeat
                        sx={{
                          fontSize: 40,
                          color: getHealthColor(dashboard?.health_indicators?.pmf_strength),
                        }}
                      />
                    </Stack>
                  </CardContent>
                </Card>
              </Grid>
              <Grid item xs={12} sm={6} md={3}>
                <Card>
                  <CardContent>
                    <Stack direction="row" justifyContent="space-between" alignItems="center">
                      <Box>
                        <Typography color="text.secondary" variant="body2">
                          Total Users
                        </Typography>
                        <Typography variant="h5">
                          {dashboard?.repeat_bookings?.total_users || 0}
                        </Typography>
                      </Box>
                      <People color="primary" sx={{ fontSize: 40 }} />
                    </Stack>
                  </CardContent>
                </Card>
              </Grid>
              <Grid item xs={12} sm={6} md={3}>
                <Card>
                  <CardContent>
                    <Stack direction="row" justifyContent="space-between" alignItems="center">
                      <Box>
                        <Typography color="text.secondary" variant="body2">
                          Acharya Churn Rate
                        </Typography>
                        <Typography
                          variant="h5"
                          sx={{
                            color: getHealthColor(dashboard?.health_indicators?.supply_stability),
                          }}
                        >
                          {formatPercentage(dashboard?.supply_health?.churn_rate || 0)}
                        </Typography>
                        <Typography variant="caption" color="text.secondary">
                          Target: &lt; 5% (Healthy)
                        </Typography>
                      </Box>
                      <People
                        sx={{
                          fontSize: 40,
                          color: getHealthColor(dashboard?.health_indicators?.supply_stability),
                        }}
                      />
                    </Stack>
                  </CardContent>
                </Card>
              </Grid>
              <Grid item xs={12} sm={6} md={3}>
                <Card>
                  <CardContent>
                    <Stack direction="row" justifyContent="space-between" alignItems="center">
                      <Box>
                        <Typography color="text.secondary" variant="body2">
                          Active Acharyas
                        </Typography>
                        <Typography variant="h5">
                          {dashboard?.supply_health?.active_acharyas || 0}
                        </Typography>
                        <Typography variant="caption" color="text.secondary">
                          of {dashboard?.supply_health?.total_acharyas || 0} total
                        </Typography>
                      </Box>
                      <People color="success" sx={{ fontSize: 40 }} />
                    </Stack>
                  </CardContent>
                </Card>
              </Grid>
            </Grid>

            {/* GMV Trend Chart */}
            <Paper sx={{ p: 3, mt: 4 }}>
              <Typography variant="h6" gutterBottom>
                GMV Trend (Last 6 Months)
              </Typography>
              <ResponsiveContainer width="100%" height={300}>
                <LineChart data={gmvTrends}>
                  <CartesianGrid strokeDasharray="3 3" />
                  <XAxis dataKey="month" />
                  <YAxis />
                  <Tooltip
                    formatter={(value) => formatCurrency(value)}
                    labelStyle={{ color: '#000' }}
                  />
                  <Legend />
                  <Line
                    type="monotone"
                    dataKey="value"
                    stroke="#1976d2"
                    strokeWidth={2}
                    name="GMV"
                  />
                </LineChart>
              </ResponsiveContainer>
            </Paper>

            {/* CAC Breakdown Table */}
            {cacData && (
              <Paper sx={{ p: 3, mt: 4 }}>
                <Typography variant="h6" gutterBottom>
                  Customer Acquisition Cost by Channel
                </Typography>
                <TableContainer>
                  <Table>
                    <TableHead>
                      <TableRow>
                        <TableCell>Channel</TableCell>
                        <TableCell align="right">CAC</TableCell>
                        <TableCell align="right">Users</TableCell>
                        <TableCell align="right">Spend</TableCell>
                      </TableRow>
                    </TableHead>
                    <TableBody>
                      {Object.entries(cacData.cac.cac_by_channel || {}).map(([channel, cac]) => (
                        <TableRow key={channel}>
                          <TableCell>{channel.replace('_', ' ').toUpperCase()}</TableCell>
                          <TableCell align="right">{formatCurrency(cac)}</TableCell>
                          <TableCell align="right">-</TableCell>
                          <TableCell align="right">-</TableCell>
                        </TableRow>
                      ))}
                      <TableRow sx={{ fontWeight: 'bold', bgcolor: 'grey.100' }}>
                        <TableCell>
                          <strong>Blended CAC</strong>
                        </TableCell>
                        <TableCell align="right">
                          <strong>{formatCurrency(cacData.summary.blended_cac)}</strong>
                        </TableCell>
                        <TableCell align="right">
                          <strong>{cacData.summary.total_users}</strong>
                        </TableCell>
                        <TableCell align="right">
                          <strong>{formatCurrency(cacData.summary.total_spend)}</strong>
                        </TableCell>
                      </TableRow>
                    </TableBody>
                  </Table>
                </TableContainer>
              </Paper>
            )}

            {/* Health Indicators Summary */}
            <Grid container spacing={3} sx={{ mt: 2 }}>
              <Grid item xs={12}>
                <Paper sx={{ p: 3 }}>
                  <Typography variant="h6" gutterBottom>
                    Business Health Indicators
                  </Typography>
                  <Grid container spacing={2}>
                    <Grid item xs={12} sm={6} md={3}>
                      <Typography variant="subtitle2" color="text.secondary">
                        GMV Growth
                      </Typography>
                      <Typography
                        variant="body1"
                        sx={{
                          color: getHealthColor(dashboard?.health_indicators?.gmv_growth),
                          textTransform: 'capitalize',
                        }}
                      >
                        {dashboard?.health_indicators?.gmv_growth || '-'}
                      </Typography>
                    </Grid>
                    <Grid item xs={12} sm={6} md={3}>
                      <Typography variant="subtitle2" color="text.secondary">
                        Product-Market Fit
                      </Typography>
                      <Typography
                        variant="body1"
                        sx={{
                          color: getHealthColor(dashboard?.health_indicators?.pmf_strength),
                          textTransform: 'capitalize',
                        }}
                      >
                        {dashboard?.health_indicators?.pmf_strength || '-'}
                      </Typography>
                    </Grid>
                    <Grid item xs={12} sm={6} md={3}>
                      <Typography variant="subtitle2" color="text.secondary">
                        Supply Stability
                      </Typography>
                      <Typography
                        variant="body1"
                        sx={{
                          color: getHealthColor(dashboard?.health_indicators?.supply_stability),
                          textTransform: 'capitalize',
                        }}
                      >
                        {dashboard?.health_indicators?.supply_stability || '-'}
                      </Typography>
                    </Grid>
                    <Grid item xs={12} sm={6} md={3}>
                      <Typography variant="subtitle2" color="text.secondary">
                        LTV:CAC Ratio
                      </Typography>
                      <Typography
                        variant="body1"
                        sx={{
                          color: getHealthColor(
                            dashboard?.health_indicators?.ltv_cac_ratio >= 3
                              ? 'excellent'
                              : 'developing'
                          ),
                        }}
                      >
                        {dashboard?.health_indicators?.ltv_cac_ratio?.toFixed(2) || '-'}
                      </Typography>
                    </Grid>
                  </Grid>
                </Paper>
              </Grid>
            </Grid>
          </>
        )}

        {/* Snackbar */}
        <Snackbar open={snackbar.open} autoHideDuration={6000} onClose={closeSnackbar}>
          <Alert onClose={closeSnackbar} severity={snackbar.severity}>
            {snackbar.message}
          </Alert>
        </Snackbar>
      </Container>
    </Layout>
  );
}

export default withAuth(InvestorMetrics, { requireAdmin: true });
